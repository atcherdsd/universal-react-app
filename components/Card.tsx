import React, { ReactNode } from 'react';
import Image from 'next/image';
import { Positions } from '../types/types';

const Card = (props: Positions): JSX.Element => {
  return (
    <div className="Card-wrapper">
      <hr className="Card-line__up"></hr>
      <div className="Card-container">
        <div className="Card-header">
          <div className="Card-title">{props.title}</div>
          <div className="Card-logo__container">
            <Image className="Card-logo" src={props.logo} width={24} height={24} alt="Brand" />
          </div>
        </div>
        <div className="Card-content__container">
          <div className="Card-image-container">
            <Image
              className="Card-image"
              src={props.img}
              width={160}
              height={160}
              alt="Smartphone"
            />
          </div>
          <div className="Card-main-content">
            <div className="Card-review__container">
              <div className="Card-review">
                <div className="Card-review-text">Expert review</div>
                <div className="Card-review-base">{props.review}</div>
              </div>
              <div className="Card-rating__container">
                <div className="Card-rating">{props.rating}</div>
                <div className="Card-rating-text">from 5</div>
              </div>
            </div>
            <div className="Card-description__container">
              <div className="Card-description">
                <ul>
                  {props.description.info.map((elem: string): ReactNode => {
                    return (
                      <li className="Card-description__item" key={elem}>
                        {elem}
                      </li>
                    );
                  })}
                </ul>
              </div>
            </div>
          </div>
          <div className="Card-price__container">
            <div className="Card-price">{props.price}</div>
            <div className="Card-stock">{props.stock}</div>
            <div className="Card-dispatch">{props.delivery}</div>
          </div>
        </div>
        <hr className="Card-line"></hr>
      </div>
    </div>
  );
};

export default Card;
