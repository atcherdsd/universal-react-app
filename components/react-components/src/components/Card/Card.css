.Card-container {
    width: 80%;
    display: flex;
    justify-content: center;
    flex-direction: column;
}

.Card-header {
    display: flex;
    flex-wrap: wrap;
    margin-bottom: 1rem;
}
.Card-title {
    width: 70%;
    text-align: left;
    color: #5281d5;
    font-size: 1.1rem;
    font-weight: bold;
}
.Card-logo__container {
    padding-left: 2rem;
}
.Card-logo {
    max-height: 1.5rem;
}

.Card-content__container {
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: space-between;
}
.Card-image {
    height: 10rem;
    vertical-align: top;
    text-align: center;
    margin-bottom: 1rem;
}
.Card-description__container {
    max-width: 30rem;
}
.Card-main-content {
    width: 45%;
}
.Card-review__container {
    background-color: #e1e9e9;
    border-radius: 0.4rem;
    padding: 0 0.5rem;
    display: flex;
    max-width: 30rem;
    justify-content: space-between;
    margin-bottom: 0.5rem;
}
.Card-review {
    text-align: left;
}
.Card-review-text {
    font-weight: bold;
}
.Card-review-base {
    color: #4c76c1;
    font-size: 1.1rem;
}
.Card-rating__container {
    display: flex;
    flex-direction: column;
    margin-left: 2rem;
    background-image: linear-gradient(180deg,#00b21c,#177412);
    border-radius: 0.4rem;
    color: #fff;
    padding: 0.3rem;
}

.Card-description {
    text-align: left;
}
.Card-price__container {
    display: flex;
    flex-direction: column;
    gap: 1rem;
}
.Card-price {
    font-size: 1.875rem;
    color: #5281d5;
}
.Card-stock {
    color: #3ba441;
    font-weight: bold;
}

.Card-line {
    border: 0;
    height: 0.1rem;
    background: linear-gradient(90deg,transparent 0,#d8d8d8 5%,#d8d8d8 95%,transparent);
    margin: 2rem;
}