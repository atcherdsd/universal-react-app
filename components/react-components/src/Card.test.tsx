import React from 'react';
import { render, screen } from '@testing-library/react';
import Card from './Card';

describe('Card component', () => {
  test('Card component renders', () => {
    render(
      <Card
        title={''}
        review={''}
        rating={''}
        price={''}
        stock={''}
        delivery={''}
        logo={''}
        img={''}
        description={{
          info: [],
        }}
      />
    );
    expect(screen.getByText('from 5')).toBeInTheDocument();
    expect(screen.getByRole('list')).toBeInTheDocument();
    expect(screen.getByRole('separator')).toBeInTheDocument();
    expect(screen.getByAltText(/brand/i)).toBeInTheDocument();
  });
  it('Card renders without data', () => {
    render(
      <Card
        title={''}
        review={''}
        rating={''}
        price={''}
        stock={''}
        delivery={''}
        logo={''}
        img={''}
        description={{
          info: [],
        }}
      />
    );
    expect(screen.queryByText('from 6')).toBeNull();
    expect(screen.queryByRole('nav')).toBeNull();
  });
  // it('Card snapshot', () => {
  //   const snapshot = render(
  //     <Card
  //       title={''}
  //       review={''}
  //       rating={''}
  //       price={''}
  //       stock={''}
  //       delivery={''}
  //       logo={''}
  //       img={''}
  //       description={{
  //         info: [],
  //       }}
  //     />
  //   );
  //   expect(snapshot).toMatchSnapshot();
  // });
});
