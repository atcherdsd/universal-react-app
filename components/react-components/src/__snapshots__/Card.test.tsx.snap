// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Card snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Card-container"
      >
        <div
          class="Card-header"
        >
          <div
            class="Card-title"
          />
          <div
            class="Card-logo__container"
          >
            <img
              alt="Brand"
              class="Card-logo"
              src=""
            />
          </div>
        </div>
        <div
          class="Card-content__container"
        >
          <div
            class="Card-image-container"
          >
            <img
              alt="Smartphone"
              class="Card-image"
              src=""
            />
          </div>
          <div
            class="Card-main-content"
          >
            <div
              class="Card-review__container"
            >
              <div
                class="Card-review"
              >
                <div
                  class="Card-review-text"
                >
                  Expert review
                </div>
                <div
                  class="Card-review-base"
                />
              </div>
              <div
                class="Card-rating__container"
              >
                <div
                  class="Card-rating"
                />
                <div
                  class="Card-rating-text"
                >
                  from 5
                </div>
              </div>
            </div>
            <div
              class="Card-description__container"
            >
              <div
                class="Card-description"
              >
                <ul />
              </div>
            </div>
          </div>
          <div
            class="Card-price__container"
          >
            <div
              class="Card-price"
            />
            <div
              class="Card-stock"
            />
            <div
              class="Card-dispatch"
            />
          </div>
        </div>
        <hr
          class="Card-line"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Card-container"
    >
      <div
        class="Card-header"
      >
        <div
          class="Card-title"
        />
        <div
          class="Card-logo__container"
        >
          <img
            alt="Brand"
            class="Card-logo"
            src=""
          />
        </div>
      </div>
      <div
        class="Card-content__container"
      >
        <div
          class="Card-image-container"
        >
          <img
            alt="Smartphone"
            class="Card-image"
            src=""
          />
        </div>
        <div
          class="Card-main-content"
        >
          <div
            class="Card-review__container"
          >
            <div
              class="Card-review"
            >
              <div
                class="Card-review-text"
              >
                Expert review
              </div>
              <div
                class="Card-review-base"
              />
            </div>
            <div
              class="Card-rating__container"
            >
              <div
                class="Card-rating"
              />
              <div
                class="Card-rating-text"
              >
                from 5
              </div>
            </div>
          </div>
          <div
            class="Card-description__container"
          >
            <div
              class="Card-description"
            >
              <ul />
            </div>
          </div>
        </div>
        <div
          class="Card-price__container"
        >
          <div
            class="Card-price"
          />
          <div
            class="Card-stock"
          />
          <div
            class="Card-dispatch"
          />
        </div>
      </div>
      <hr
        class="Card-line"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
